### PR TITLE
[@types/vscode] Add 1.14 Extension API

### PR DESCRIPTION
**Pull request by octref**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] 🚨 If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package. I didn't do this because similar to how [`@types/chrome`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chrome/index.d.ts) matches [Chrome API](https://developers.chrome.com/extensions/api_index), `@types/vscode` should match the VS Code API described at https://code.visualstudio.com/api/references/vscode-api. The `vscode` NPM module contain a JS script to pull `vscode.d.ts` from microsoft/vscode manually, and we'll deprecate that package. See https://github.com/Microsoft/vscode/issues/70175 for details.
- [ ] 🚨 Create it with `dts-gen --dt`, not by basing it on an existing project. This is built upon https://github.com/Microsoft/vscode/blob/release/1.14/src/vs/vscode.d.ts.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

---

Questions:

- Is this header OK? `// Type definitions for VS Code 1.14 extension development`, or should I change it to `// Type definitions for VS Code 1.14` for the versioning to work correctly?

### CODE REVIEW

File: types/vscode/tslint.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
+{
+	"extends": "dtslint/dt.json",
+	"rules": {
+		"adjacent-overload-signatures": false,
+		"array-type": false,
+		"arrow-return-shorthand": false,
+		"ban-types": false,
+		"callable-types": false,
+		"comment-format": false,
+		"dt-header": false,
```
Comment (weswigham) [inline types/vscode/tslint.json]: You cannot disable this rule (as you suspected) - publishing won't work if the header metadata isn't readable.

**Review by weswigham — CHANGES_REQUESTED** [review]

File: types/vscode/tslint.json (hunk 1 above)
Comment (octref) [reply types/vscode/tslint.json]: Thanks, resolved.

Is that the only field I should care about for publishing, or are there other fields affecting publishing too?

Comment (weswigham) [reply types/vscode/tslint.json]: Well, you need the lint rule to actually pass~

Comment (octref) [reply types/vscode/tslint.json]: Oops, sorry. Stupid me.

**Review by weswigham — APPROVED** [review]

**Pull request merged**